add node config

nixbitcoin opened the pull request:
I decided to open source the code of the [nixbitcoin.org](https://nixbitcoin.org) website/demo node. I hope that open sourcing it will allow it to serve as a model nix-bitcoin deployment, clean up the code and decentralize control.

I currently have the secrets on a separate unpushed branch, but I could work on converting the secrets directory to a password store, encrypt it to @erikarvstedt's, @jonasnick's, my PGP keys, and add it to this branch. I can also add your SSH keys to the root `openssh.authorizedKeys.keys`.

Also note, that the webdesign isn't in its final stage and could be completely changed.

erikarvstedt commented:
[Improvements](https://github.com/erikarvstedt/nixbitcoin.org/commits/nixbitcoin.org-ea). You can squash these if you like.

The nix-shell now supports some helpful debug commands:
```
# Run node in a container
container

# Launch the node in a container, render the website as text, shutdown node
website

# Run node in a VM.
# Hint: Run command 'q' in the VM for instant poweroff.
vm
```

nixbitcoin commented:
Pulled in your fix and
- replaced `nix-bitcoin-org` with `website` in `deploy.nix`
- proxyPass to all necessary btcpayserver subdirectories with nginx regexp
- amended `index.html` with new features

erikarvstedt commented on website/default.nix:
```diff
@@ -0,0 +1,115 @@
+    services.nginx = let
+      hostConfig = {
+        root = "/var/www";
+        locations."~ \"/btcpayserver/(apps|main|vendor|i|bundles|checkout|locales|)\"" = {
```
Suggestion (The Github suggestion feature is unavailable for me right now): 
```
locations."~ /btcpayserver/(apps|main|vendor|i|bundles|checkout|locales)/"
```
The previous regexp had no limiting effect because it matched any URL with prefix `/btcpayserver/`:
`(...|)` matched the empty string and the final `/` was missing.

How did you find all the public sub-paths `apps|main|vendor|i|bundles|checkout|locales`?

nixbitcoin replied:
> How did you find all the public sub-paths `apps|main|vendor|i|bundles|checkout|locales`?

By going through the checkout process and watching `journalctl -fu nginx`. All these came up as errors because nginx was looking for them in `/var/www`.

erikarvstedt commented on website/default.nix:
```diff
@@ -0,0 +1,115 @@
+        locations."~ \"/btcpayserver/(apps|main|vendor|i|bundles|checkout|locales|)\"" = {
+          proxyPass = "http://${serviceAddress "btcpayserver"}";
+        };
+        locations."~ \"/(fonts|humans.txt|robots.txt|ads.txt)\"" = {
```
Why do we need this?

nixbitcoin replied:
They were also coming up as error because nginx was looking for them in `/var/www`.

erikarvstedt replied:
But btcpayserver doesn't host any of these files, so the error is just delegated to btcpayserver. I'd remove this setting.

nixbitcoin replied:
Fixed in d113e8247270e5c92105d4cf72bcd77c02b3ad18

nixbitcoin commented:
Had to add some more locations

erikarvstedt commented:
[Fixups](https://github.com/erikarvstedt/nixbitcoin.org/commits/nixbitcoin.org-ea-2).

erikarvstedt commented:
I've just noticed that we should use the [transparent logo](https://pbs.twimg.com/profile_images/1346032831198355457/4ygcw6dg_400x400.png) for https://github.com/fort-nix.

nixbitcoin commented:
> I've just noticed that we should use the [transparent logo](https://pbs.twimg.com/profile_images/1346032831198355457/4ygcw6dg_400x400.png) for https://github.com/fort-nix.

Paging @jonasnick

nixbitcoin commented:
Squashed. I moved the `/btcpayserver/` location from the `extraConfig` to the `services.nginx.virtualHosts.<name>.locations` because it then automatically includes the `nginx-recommended-proxy-headers.conf`.

erikarvstedt commented:
Could you add https://github.com/erikarvstedt/nixbitcoin.org/commit/8c7d0dfdc35c3a76b3a1a2f596f85a3c376f7bcf? This fixes the mix of tabs and spaces.

nixbitcoin commented:
> Could you add [erikarvstedt@8c7d0df](https://github.com/erikarvstedt/nixbitcoin.org/commit/8c7d0dfdc35c3a76b3a1a2f596f85a3c376f7bcf)? This fixes the mix of tabs and spaces.

Done

jonasnick commented:
> I've just noticed that we should use the [transparent logo](https://pbs.twimg.com/profile_images/1346032831198355457/4ygcw6dg_400x400.png) for https://github.com/fort-nix.

I changed it to the transparent logo, but we may have already used that one because I'm not seeing a difference.

erikarvstedt commented:
@jonasnick, you're right, Github adds a CSS `background-color` to the logo.

nixbitcoin commented:
@erikarvstedt thank you for your review. Very helpful.

@jonasnick How do you use https://github.com/bitcoin-core/bitcoin-maintainer-tools/blob/master/github-merge.py without including pull request descriptions?

jonasnick commented:
@nixbitcoin I don't remember anything about this, but apparently I have locally replaced `body = info['body'].strip()` with `body = ""`.
